### PR TITLE
ddl/ingest: set `minCommitTS` when detect remote duplicate keys

### LINKED ISSUE #55587: Cannot add unique index after flashback cluster
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
set @@global.tidb_gc_life_time = '100m';

drop table if exists t;
create table t (a int, b int);
insert into t values (1, 1);

select now();  -- @current_time
select sleep(3);
alter table t add index idx(b);
flashback cluster to timestamp '2024-08-22 11:36:36'; -- @current_time
alter table t add unique index idx(b);
```

### 2. What did you expect to see? (Required)

No error.

### 3. What did you see instead (Required)

```
ERROR 1105 (HY000): meet key error in duplicate detect response: [src/import/[duplicate_detect.rs:88](http://duplicate_detect.rs:88/)]: found a Delete key with commits ts 451995312167845889 larger than min_commit_ts of importer 0
```

### 4. What is your TiDB version? (Required)

9690dab33

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #55587

Problem Summary:

https://github.com/pingcap/tidb/issues/55587#issuecomment-2303941700

### What changed and how does it work?

Set `minCommitTS` when detect remote duplicate keys

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
    After this PR:
    ```sql
    mysql> set @@global.tidb_gc_life_time = '100m';
    Query OK, 0 rows affected (0.01 sec)

    mysql> drop table if exists t;
    Query OK, 0 rows affected, 1 warning (0.00 sec)

    mysql> create table t (a int, b int);
    Query OK, 0 rows affected (0.10 sec)

    mysql> insert into t values (1, 1);
    Query OK, 1 row affected (0.01 sec)

    mysql> select now();
    +---------------------+
    | now()               |
    +---------------------+
    | 2024-09-05 17:31:47 |
    +---------------------+
    1 row in set (0.00 sec)

    mysql> alter table t add index idx(b);
    Query OK, 0 rows affected (2.20 sec)

    mysql> flashback cluster to timestamp '2024-09-05 17:31:47';
    Query OK, 0 rows affected (2.22 sec)

    mysql> alter table t add unique index idx(b);
    Query OK, 0 rows affected (2.24 sec)
    ```
    Because using flashback cluster in real tikv test is very unstable, I test it manually. There is also an integration test on endless.
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ingest/backend_mgr.go [hunk 1]
```diff
@@ -153,7 +154,13 @@ func (m *litBackendCtxMgr) Register(
 		return nil, err
 	}
 
-	bcCtx := newBackendContext(ctx, jobID, bd, cfg, defaultImportantVariables, m.memRoot, m.diskRoot, etcdClient)
+	physical, logical, err := bd.GetTS(ctx)
+	if err != nil {
+		return nil, err
+	}
+	curTS := oracle.ComposeTS(physical, logical)
```
Comment (lance6716) [inline pkg/ddl/ingest/backend_mgr.go:161]: this value should be saved in checkpoint, so we won't skip checking when failover

Comment (lance6716) [reply pkg/ddl/ingest/backend_mgr.go:161]: maybe use DDL job RealStartTS?

Comment (tangenta) [reply pkg/ddl/ingest/backend_mgr.go]: https://github.com/pingcap/tidb/pull/55588/commits/1eacf7d510730d8cb635700b77338ab576d8d565

**Review by lance6716 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**